controllers/azurecluster_reconciler_test.go

```diff
@@ -30,72 +30,107 @@ import (
 	"sigs.k8s.io/cluster-api-provider-azure/azure/scope"
 	"sigs.k8s.io/cluster-api-provider-azure/azure/services/resourceskus"
 	gomockinternal "sigs.k8s.io/cluster-api-provider-azure/internal/test/matchers/gomock"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
-type expect func(grp *mock_azure.MockReconcilerMockRecorder, vnet *mock_azure.MockReconcilerMockRecorder, sg *mock_azure.MockReconcilerMockRecorder, rt *mock_azure.MockReconcilerMockRecorder, sn *mock_azure.MockReconcilerMockRecorder, natg *mock_azure.MockReconcilerMockRecorder, pip *mock_azure.MockReconcilerMockRecorder, lb *mock_azure.MockReconcilerMockRecorder, dns *mock_azure.MockReconcilerMockRecorder, bastion *mock_azure.MockReconcilerMockRecorder, peer *mock_azure.MockReconcilerMockRecorder)
+func TestAzureClusterServiceReconcile(t *testing.T) {
+	cases := map[string]struct {
+		expectedError string
+		expect        func(one *mock_azure.MockReconcilerMockRecorder, two *mock_azure.MockReconcilerMockRecorder, three *mock_azure.MockReconcilerMockRecorder)
```

Comment:
Is there any significance in having three services in the test cases? Or is it just for the sake of convenience so it'll cover most cases?

Reply:
no particular significance just chose a number big enough that it covers the basic edge cases (first, last, middle) but not too big that it makes the tests very long/repetitive and hard to read. 

Here since we're testing code with a for loop, the behavior should be the same whether there are 3, 4, 5, or 50 services. The goal is not to test the behavior of the for loop itself but more that for a given number of items our code is behaving as expected. I chose naming "one" "two" "three" since it makes the ordering aspect of the test very obvious and easy to follow (services are reconciled in order and deleted in reverse).